### PR TITLE
Added Speed Status Bar #855

### PR DESCRIPTION
**Pull request by Westsi**
In order to solve the issue #855, I implemented a speed status bar in the way that was discussed in that issue (r.e. speed calculations), and added it to all the same places as the existing health, mana, defence, and experience bars. I created a texture for it. 

I have tested the bar and most of the exceptions to the 400 speed cap in game. I have not tested the racing helmet nor the black cat pet as I don't have the money for that, but since they work in the same way as the other exceptions I see no reason why they shouldn't work.

I tested:
 - Cactus Knife (only in garden)
 - Snail pet
 - Young Dragon Armour

### CODE REVIEW

**Review by viciscat — APPROVED** [review]
It works well and the code looks pretty good so uh good job 👍 
Did not test the max speed exceptions tho.

File: src/main/java/de/hysky/skyblocker/config/configs/UIAndVisualsConfig.java [hunk 1]
```diff
@@ -179,6 +179,8 @@ public static class OldBarPositions {
 
         @SerialEntry
         public OldBarPosition experienceBarPosition = OldBarPosition.LAYER1;
+        @SerialEntry
+        public OldBarPosition speedBarPosition = OldBarPosition.LAYER1;
```
Comment (kevinthegreat1) [inline src/main/java/de/hysky/skyblocker/config/configs/UIAndVisualsConfig.java:184]: This is for legacy reasons, not needed here.
```suggestion
        public OldBarPosition experienceBarPosition = OldBarPosition.LAYER1;
```

File: src/main/java/de/hysky/skyblocker/skyblock/fancybars/FancyStatusBars.java [hunk 2]
```diff
@@ -76,6 +79,8 @@ public static void init() {
         loadOldBarPosition(defense, counts, barPositions.defenceBarPosition);
         StatusBar experience = statusBars.get("experience");
         loadOldBarPosition(experience, counts, barPositions.experienceBarPosition);
+        StatusBar speed = statusBars.get("speed");
+        loadOldBarPosition(speed, counts, barPositions.speedBarPosition);
```
Comment (kevinthegreat1) [inline src/main/java/de/hysky/skyblocker/skyblock/fancybars/FancyStatusBars.java:83]: Same as above.
```suggestion
```

**Review by kevinthegreat1 — CHANGES_REQUESTED** [review]
Code looks great; just one thing.

File: src/main/java/de/hysky/skyblocker/config/configs/UIAndVisualsConfig.java (hunk 1 above)
Comment (viciscat) [reply src/main/java/de/hysky/skyblocker/config/configs/UIAndVisualsConfig.java:184]: I think it's kinda needed? for default positioning, unless i don't remember my own code

Comment (kevinthegreat1) [reply src/main/java/de/hysky/skyblocker/config/configs/UIAndVisualsConfig.java]: Woah I messed up. I will fix.

**Review by kevinthegreat1 — DISMISSED** [review]
Works well. Except this is what new players see lol.
Screenshot 2024-08-22 at 02 10 17

**Comment by Westsi** [comment]
I looked at that but I don't think that it's any bigger of an issue than it was before this change. I could change the default positioning of it, but at that point why not change the defaults of all of them?

**Comment by kevinthegreat1** [comment]
Yeah. I'm just leaving that here in case someone wants to modify the defaults for all the bars. We can merge this pr already.

**Review by kevinthegreat1 — APPROVED** [review]
Changed the defaults a bit.

**Comment by kevinthegreat1** [comment]
Thanks for contributing!

**Pull request merged**